application/akka/scala/example/src/main/scala/com/example/Example.scala

```diff
@@ -0,0 +1,78 @@
+package com.example
+
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.scaladsl.Behaviors
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+
+import scala.concurrent.ExecutionContextExecutor
+import scala.io.StdIn
+import scala.util.Failure
+import scala.util.Success
+
+//#main-class
+object Example {
+  //#start-http-server
+  private def startHttpServer(routes: Route)(implicit system: ActorSystem[_]): Unit = {
+    // Akka HTTP still needs a classic ActorSystem to start
+    import system.executionContext
+
+    val futureBinding = Http().newServerAt("localhost", 8080).bind(routes)
+    futureBinding.onComplete {
+      case Success(binding) =>
+        val address = binding.localAddress
+        system.log.info("Server online at http://{}:{}/", address.getHostString, address.getPort)
+      case Failure(ex) =>
+        system.log.error("Failed to bind HTTP endpoint, terminating system", ex)
+        system.terminate()
+    }
+  }
+  //#start-http-server
+  def main(args: Array[String]): Unit = {
+    //#server-bootstrapping
+    val rootBehavior = Behaviors.setup[Nothing] { context =>
+      val route =
+        path("") {
+          get {
+            if (sys.env.getOrElse("DEMO_USERNAME", "").isEmpty || sys.env.getOrElse("DEMO_PASSWORD", "").isEmpty) {
+              complete("not all variables are set\n")
+            } else {
+              complete("Welcome " + sys.env("DEMO_USERNAME") + "\n")
+            }
+          }
+        }
+      startHttpServer(route)(context.system)
+
+      Behaviors.empty
+    }
+    val system = ActorSystem[Nothing](rootBehavior, "HelloAkkaHttpServer")
+    //#server-bootstrapping
+  }
+
+  // TODO Select the code above or the commented one
```

Comment:
First of all, I would refrain from giving the user choices like this one, unless they provide valuable information.

If you think providing this alternative code is valuable then please explain this choice somewhere in the readme, to make it clear for the user which option is (better) for which use-case.

Moreover, if we want to keep this alternative, I strongly suggest finding a different way of presenting it, e.g. putting it in a different file, having two scala and two java examples, etc. I think commented code is not the best way to go about this.

Reply:
This is not for the user, it is for us.
I was oscilating between the two versions of the code and I didn't know which one to pick. Therefore, I put them both and added the todo to choose.
Let's have a small chat about it and I will adjust accordingly.

Reply:
Based on the discussion, I removed the commented code, since it is not as elegant as the other one.
Also, the one above the TODO is similar to the one provided by the template, which means the user can easily expand upon it.